client/reader/stream/index.jsx

```diff
@@ -133,6 +116,7 @@ class ReaderStream extends React.Component {
 		if ( this.props.shouldCombineCards ) {
 			items = combineCards( items );
 		}
+		items = map( items, this.props.transformStreamItems );
```

Comment:
shouldn't this happen first? it's going to overwrite every item with `isRecommended: true` for the search stream. We get away with it because we don't pass a recommendations store and shouldCombineCards is false for the search store, but I think it makes things a bit less extensible to do it last.